api/src/main/java/net/kyori/adventure/util/ComponentMessageThrowable.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * This file is part of adventure, licensed under the MIT License.
+ *
+ * Copyright (c) 2017-2020 KyoriPowered
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package net.kyori.adventure.util;
+
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.TextComponent;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+/**
+ * An extension interface for {@link Throwable}s to provide a {@link Component}-based message.
+ */
+public interface ComponentMessageThrowable {
+  /**
+   * Gets the {@link Component}-based message from a {@link Throwable}, if available.
+   *
+   * @param throwable the throwable
+   * @return the message
+   */
+  static @Nullable Component get(final @Nullable Throwable throwable) {
```

Comment:
Copy-pasta from Discord:
I don't think I like the sound of `ComponentMessageThrowable.get(Throwable)` -> `Component`?
static "get" usually implies it returns the type of the containing class
would prefer `getMessage(Throwable)`
and `getOrConvertMessage(Throwable)`